sale_product_pack/models/sale_order.py

```diff
@@ -37,3 +37,19 @@ def check_pack_line_unlink(self):
                     " delete the pack itself"
                 )
             )
+
+    def write(self, vals):
+        if "order_line" in vals:
+            to_delete_ids = [e[1] for e in vals["order_line"] if e[0] == 2]
+            subpacks_to_delete_ids = (
+                self.env["sale.order.line"]
+                .search(
+                    [("id", "child_of", to_delete_ids), ("id", "not in", to_delete_ids)]
+                )
+                .ids
+            )
+            if subpacks_to_delete_ids:
+                for cmd in vals["order_line"]:
```

Comment:
What happens if one of the `subpacks_to_delete_ids` is not in the command line? That one won't get removed. You need to assure that all needed `(2, x)` commands are there.